packages/graph-engine/src/programmatic/input.ts

```diff
@@ -135,10 +152,15 @@ export class Input<T = any> extends Port<T> {
 		const type = this.fullType();
 		const serialized = {
 			name: this.name,
-			value: this.value,
 			type: type.type
 		} as SerializedInput;
 
+		// Only include value in serialization if not a connected non-variadic input
+		// This prevents redundant data storage in the serialized format
+		if (!this.isConnected || this.variadic) {
```

Comment:
We should probably optimized the `isConnected` function here as it currently is trying to get the full array could and then do a comparison, we only need to know if at miniumum one connection exists like `return !!this._edges[0];`